x-pack/plugins/ml/public/application/components/scatterplot_matrix/scatterplot_matrix_vega_lite_spec.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// There is still an issue with Vega Lite's typings with the strict mode Kibana is using.
+// @ts-ignore
+import type { TopLevelSpec } from 'vega-lite/build-es5/vega-lite';
+
+import {
+  euiPaletteColorBlind,
+  euiPaletteGray,
+  euiPaletteNegative,
+  euiPalettePositive,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+export const LEGEND_TYPES = {
+  NOMINAL: 'nominal',
+  QUANTITATIVE: 'quantitative',
+} as const;
+export type LegendType = typeof LEGEND_TYPES[keyof typeof LEGEND_TYPES];
+
+export const OUTLIER_SCORE_FIELD = 'outlier_score';
+
+const SCATTERPLOT_SIZE = 125;
+
+const DEFAULT_COLOR = euiPaletteColorBlind()[0];
+const COLOR_OUTLIER = euiPaletteNegative(2)[1];
+const COLOR_GRAY = euiPaletteGray(5)[0];
+const COLOR_RANGE_NOMINAL = euiPaletteColorBlind({ rotations: 2 });
+const COLOR_RANGE_QUANTITATIVE = euiPalettePositive(5);
+
+const getColorSpec = (outliers = true, color?: string, legendType?: LegendType) => {
+  if (outliers) {
+    return {
+      condition: {
+        value: COLOR_OUTLIER,
+        test: `(datum['${OUTLIER_SCORE_FIELD}'] >= mlOutlierScoreThreshold.cutoff)`,
+      },
+      value: COLOR_GRAY,
+    };
+  }
+
+  if (color !== undefined && legendType !== undefined) {
+    return {
+      field: color,
+      type: legendType,
+      scale: {
+        range: legendType === LEGEND_TYPES.NOMINAL ? COLOR_RANGE_NOMINAL : COLOR_RANGE_QUANTITATIVE,
+      },
+    };
+  }
+
+  return { value: DEFAULT_COLOR };
+};
+
+export const getScatterplotMatrixVegaLiteSpec = (
+  values: any[],
+  columns: string[],
+  resultsField?: string,
+  color?: string,
+  legendType?: LegendType,
+  dynamicSize?: boolean
+): TopLevelSpec => {
+  const outliers = resultsField !== undefined;
+  const transform = columns.map((column) => ({
+    calculate: `datum['${column}']`,
+    as: column,
+  }));
+  transform.push({
+    calculate: `datum['${resultsField}.${OUTLIER_SCORE_FIELD}']`,
+    as: OUTLIER_SCORE_FIELD,
+  });
+
+  return {
+    $schema: 'https://vega.github.io/schema/vega-lite/v4.8.1.json',
+    padding: 10,
+    config: {
+      axis: {
+        domainColor: '#ccc',
```

Comment:
Can you use EUI colors here? Will these work with dark theme?

Reply:
Fixed in c6389327223. I fixed dark theme too and it mostly works besides a glitch. I added a comment in the code about it and added an item to the meta issue.